maven-plugin/pom.xml

```diff
@@ -65,6 +65,29 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-invoker-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>integration-test</id>
+            <goals>
+              <goal>install</goal>
+              <goal>run</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <cloneProjectsTo>${project.build.directory}/it</cloneProjectsTo>
+          <localRepositoryPath>${project.build.directory}/local-repo</localRepositoryPath>
+          <!-- See http://maven.apache.org/plugins/maven-invoker-plugin/examples/fast-use.html -->
+          <settingsFile>src/it/settings.xml</settingsFile>
+          <scriptVariables>
+            <projectversion>${project.version}</projectversion>
+          </scriptVariables>
+          <skipInvocation>${skipTests}</skipInvocation>
+        </configuration>
+      </plugin>
```

Comment:
here we add the invoker plugin during the build of your plugin.

It runs in two times:

1. It will copy your plugin's jar (and its dependencies) build during the previous package phase in its dedicated local repository (specified by the _localRepositoryPath_), to avoid polluting yours (_~/.m2/repository_ by default)
2. It will consider every folder in _src/it_ as a maven project, and run it (after having cloned it in the _cloneProjectsTo_ directory, to avoid modification of source files), with the settings in _settingsFile_ (not merged by default with yours, _~/.m2/settings_, but [you can](http://maven.apache.org/plugins/maven-invoker-plugin/run-mojo.html#mergeUserSettings) if you want)